tests/Microsoft.Identity.Test.Unit/Throttling/ThrottlingAcceptanceTests.cs

```diff
@@ -203,7 +203,29 @@ public async Task RetryAfter_ConfidentialClient_Async()
             }
         }
 
+        [TestMethod]
+        public async Task RetryAfter_ConfidentialClient_ErrorMessage_Async()
+        {
+            using (var httpManagerAndBundle = new MockHttpAndServiceBundle())
+            {
+                var httpManager = httpManagerAndBundle.HttpManager;
+                var app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
+                                                              .WithClientSecret(TestConstants.ClientSecret)
+                                                              .WithHttpManager(httpManager)
+                                                              .BuildConcrete();
 
+                httpManager.AddInstanceDiscoveryMockHandler();
+                var tokenResponse = httpManager.AddMockHandlerForThrottledResponseMessage();
+
+                var ex = await AssertException.TaskThrowsAsync<MsalServiceException>(
```

Comment:
What happens if you run this test without having added the custom error handling code ? Does it reproduce the exception observed by the customer?

Reply:
Yes. It does when adding AddMockHandlerForThrottledResponseMessage()